go/matrix/matrix.go

```diff
@@ -0,0 +1,100 @@
+// This package implements matrix parsing utilities, to generate arrays
+// from string representations of matrices.
+package matrix
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+const ROW_SEPARATOR string = "\n"
+const COLUMN_SEPARATOR string = " "
+
+//  struct representation of a matrix
+type matrix struct {
+	values [][]int
+}
+
+//  New creates a new matrix object based on a string representation of
+//  said matrix. This representation contains one or more non empty
+//  lines, each containing the same number of integer values, separated
+//  by spaces.
+func New(matrixString string) (*matrix, error) {
+	var numCols int = 0
+	var err error = nil
+
+	rows := strings.Split(matrixString, ROW_SEPARATOR)
+	var numRows int = len(rows)
+	var values = make([][]int, numRows)
+
+	for rowInd, row := range rows {
+		row = strings.Trim(row, " ")
+		rowCells := strings.Split(row, COLUMN_SEPARATOR)
+		if rowInd == 0 {
+			numCols = len(rowCells)
+			if numCols == 0 { // Check first line's dimension
+				err = errors.New("Empty first row")
+			}
+		} else if len(rowCells) != numCols { // Check line size consistency
+			err = errors.New("Inconsistent line numbers")
+		}
+
+		if err == nil {
```

Comment:
following what happens with `err` while not difficult per se, for me is not needed. Return as soon as you have identified an error situation and the code will be simpler.